### PR TITLE
Changes to bone repairing

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

Bone glue is limited, with 5% glue used per repair
Repairing bones with a screwdriver costs 2 metal sheets per repair
Adds a machine to the USS Almayer that allows you to refill bone glue

# Explain why it's good for the game

By limiting bone glue, you will limit the amount of surgery that happens groundside without sacrificing other resources that have important use, or by having additional supplies delivered. 
If there is a limit to how much surgery can be done, it will force the groundside surgeon to prioritize and consider the severity of the injuries, instead of fully repairing every bone.
This will increase the consequences from getting injured and having broken bones, by requiring you to either work with the downside via splints, or to return shipside.
There should be no changes to shipside surgery, as the machine should be close enough to replenish without having any problems.

# Changelog

:cl:
balance: Repairing bones with bone glue costs a portion of the bone glue's resources
balance: Repairing bones with a screwdriver costs metal
del: Removed bone glue from the medilathe
spellcheck: New description for bone glue
/:cl:

### CODE REVIEW

**Comment by Steelpoint** [comment]
Does this affect Predator bonegel surgical tools? If so, you may want to add a refiller to their ship. Very edge case scenario admittedly.

File: code/modules/surgery/bones.dm
Comment (Cthulhu80) [inline code/modules/surgery/bones.dm:60]: If bone gel becomes scarce I can definitely see a few doctors getting annoyed after a marine pushes them during surgery or for whatever reason the action is cancelled and they lose their last bone gel, as it stands the metal and or bone gel is removed here and not once the process is complete from either a failed procedure or successful one. Not saying it's need to change but just something to think about.

**Comment by kooarbiter** [comment]
5 percent seems like a lot for a tool you have to raid surgery to get any more of, I dont think you can get bonegel by itself via cargo or research.

**Comment by SarahJohnson489** [comment]
Bruhv, do you even perceive how easily bones get broken even through heavy armor? Helmet and extremities which have shitall in the face of combat castes? Do you even remember morrow going sicko mode on head breaks and brain damage effects? Unsplinted IB chance and bicard od IB removal? How a shipside med trip can take a half hour or more. Which can be even worse in lowpop which gets tied up dealing with a stream of larva removals? Getting trapped shipside because transport is holding for fob siege. Staring at the hangar for fifteen minutes because comms have long since been down and no one with a phone can get anyone with signal flares or binos. Why do you think round removing a marine over a two tap is any benefit?
This is straight up griefing. Please. I begging you. Explain how this helps. Show how many lowpop surgeon hours you have to justify that YOU will be Sisyphus shouldering the burden of what you're inflicting on everyone else. Explain why this warrants griefing literally hundreds of people for no benefit at all. Take a step back and think sweet jesus christ.

**Comment by Cthulhu80** [comment]
> Bruhv, do you even perceive how easily bones get broken even through heavy armor? Helmet and extremities which have shitall in the face of combat castes? Do you even remember morrow going sicko mode on head breaks and brain damage effects? Unsplinted IB chance and bicard od IB removal? How a shipside med trip can take a half hour or more. Which can be even worse in lowpop which gets tied up dealing with a stream of larva removals? Getting trapped shipside because transport is holding for fob siege. Staring at the hangar for fifteen minutes because comms have long since been down and no one with a phone can get anyone with signal flares or binos. Why do you think round removing a marine over a two tap is any benefit?
> 
> This is straight up griefing. Please. I begging you. Explain how this helps. Show how many lowpop surgeon hours you have to justify that YOU will be Sisyphus shouldering the burden of what you're inflicting on everyone else. Explain why this warrants griefing literally hundreds of people for no benefit at all. Take a step back and think sweet jesus christ.
> 
> 

Sarah we have talked about this, you're not you when you're hungry. Have a snickers.

**Comment by SarahJohnson489** [comment]
It's griefing hundreds of people. It's straight up round removing people in lowpop over a two-tap head break on a single push out of the round. It's insanity to do that to people for no benefit. Are both of you doing to be in every single lowpop all year every year speedrunning larva removals so fast that people don't get stuck behind the larva queue forever? We both know you're not when people like me get stuck and keep getting stuck when we make the mistake of going on the alamayer's wild ride. There isn't even autodoc fracture repairs anymore to help with the swells. And wouldn't you know it? The jerks that did that aren't around lowpop fixing fractures. WOW! Never saw the players being left holding the coder grief bag coming! I don't know if you're somehow unaware of the problems even after being informed or you just don't care about the 30-50-end of round min trip to shipside med for no benefit. Jesus christ.

Have you ever been waiting in med for 50 minutes just pondering the mysteries of how things came to this and thought wow, it's so great more people should get removed from the round? Is not making terrible situations worse so alien a thing to ask for?

**Review by Nanu308 — APPROVED** [review]
mapping lgtm

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items/tools/surgery_tools.dm
Comment (fira) [inline code/game/objects/items/tools/surgery_tools.dm:218]: stray space at start

File: code/game/objects/items/tools/surgery_tools.dm
Comment (fira) [inline code/game/objects/items/tools/surgery_tools.dm:226]: Isn't that meant to be break here rather than continue? You'd just keep on trying to refill forever if you move away

**Review by fira — CHANGES_REQUESTED** [review]

**Review by fira — APPROVED** [review]

**Review by Drulikar — CHANGES_REQUESTED** [review]
Code looks fine and the usage of bone gel seems to be fine except for the usage of metal as an alternative. At the moment it just gobbles up metal without progressing and its not clear what tool you're supposed to be using in your main hand to be using metal. You're either going to get erronous messages about wrong tool and it uses the metal. Or you try to wake them up and it uses the metal. Metal probably should just be main hand and as an alternative tool for the surgery, or you use it with the bonegel in main hand, or some other tool. 
metal

**Review by Drulikar — APPROVED** [review]

**Comment by private-tristan** [comment]
you can spamclick the machine to cause it to refill multiple times. (please just make refilling faster)

https://github.com/cmss13-devs/cmss13/assets/54422837/fa226782-3706-4962-9f4d-237a9ce62879

also, bone gel is used up when the step STARTS instead of when it ends, meaning that if you're interrupted (pushed a tile), that bone gel charge is gone. please consider moving the usage of hthe bone gel to the end of the step.

**Comment by Drulikar** [comment]
Above issues looks resolved and I don't see any issues with the change.

**Comment by Staykeu** [comment]
This is ridiculous. Under these changes it is more effective to amputate and replace a person's limb. This seems like it only exists to make an already tedious job more tedious.

**Comment by AlakRacc** [comment]
Hello. I am writing on the behalf of dozens of community members that have expressed disinterest in this PR, myself included. I would like to ask you to take a moment and consider the following.

Why would this PR be good for the game? How does it improve gameplay?

Metal is a **very vital** resource for marines, mainly for making barricades, as well as other structures but it also has other uses. Why should a resource that is so vital be used to fix bones? 

Does this change actually make a positive impact? At the moment, it seems to just incentivize other options, such as Marines opting to have their limbs replaced with cybernetics, as they only cost a few metal to produce and will not result in fractures. 

Lastly, you had mentioned wanting ‘logistics’ for groundside surgery. I would personally recommend talking to players or making a forum post asking what sort of logistical changes they would like to see.

Have a good day!

**Comment by Mister-moon1** [comment]
On advise of frozen, I will leave my opinion here

To start the scarcity of bone gel I feel is an unnecessary change which really drags the PR down, making people consider reserving space for bone gel would be a better change rather than forcing them to go through a tedious system which would need to be repeated multiple times a round.

A personal experience of the scarcity would be when someone took a bone gel from medbay during a round with a large amount of infected people, leading to the medbay, although having a resounding amount of staff to perform surgery, lack the tools to follow through.

Another point which has been brought up is the fact that making bone gel as scarce as is and needing metal for the "ghetto" method, means that it is more effective metal wise to amputate a limb and replace it with a robotic one, this would be considered LRP before the changes, considering the fact there was no realistic reason to do so, but if it is 10 times easier to maintain then I honestly cannot blame them

Overall for these changes to work, I feel like it should be less taxing overall to perform a simple mechanic, starting off with adding the bone gel to the medilathe, a spare bone gel in the field surgery cupboard, and for the ghetto method to use 2 metal rods rather than plates as it makes more sense and doesn't cripple engineers entirely for simple repairs

To put into comparison of the groundside doctor having 2 tubs of bone gel, the average person seeking surgery has 3 fractures or so, one of the most common fracture locations is the chest which if organ repair is performed which it often is, then 4 uses of the bone gel is needed to heal back to full health, this leads to just 10 people before the doctor needs a complete refill, it also allows him to continue surgery during the refill, and allows sustainability during times such as an FOB siege which will require the doctor to be vigilant of his resources

**Comment by Drulikar** [comment]
> On advise of frozen, I will leave my opinion here
> 
> To start the scarcity of bone gel I feel is an unnecessary change which really drags the PR down, making people consider reserving space for bone gel would be a better change rather than forcing them to go through a tedious system which would need to be repeated multiple times a round.
> 
> A personal experience of the scarcity would be when someone took a bone gel from medbay during a round with a large amount of infected people, leading to the medbay, although having a resounding amount of staff to perform surgery, lack the tools to follow through.
> 
> Another point which has been brought up is the fact that making bone gel as scarce as is and needing metal for the "ghetto" method, means that it is more effective metal wise to amputate a limb and replace it with a robotic one, this would be considered LRP before the changes, considering the fact there was no realistic reason to do so, but if it is 10 times easier to maintain then I honestly cannot blame them
> 
> Overall for these changes to work, I feel like it should be less taxing overall to perform a simple mechanic, starting off with adding the bone gel to the medilathe, a spare bone gel in the field surgery cupboard, and for the ghetto method to use 2 metal rods rather than plates as it makes more sense and doesn't cripple engineers entirely for simple repairs, the doctor will also need to commit space to said metal rods should they want convienience, with a bad trade off being made if their mind is changed and want the sheets back
> 
> To put into comparison of the groundside doctor having 2 tubs of bone gel, the average person seeking surgery has 3 fractures or so, one of the most common fracture locations is the chest which if organ repair is performed which it often is, then 4 uses of the bone gel is needed to heal back to full health, this leads to just 10 people before the doctor needs a complete refill, it also allows him to continue surgery during the refill, and allows sustainability during times such as an FOB siege which will require the doctor to be vigilant of his resources

This is much better feedback; but this is what you should have submitted to https://forum.cm-ss13.com/w/pr-feedback - not what you actually submitted there; nor with all the discussions *everywhere*. The github is more for implementation/coding feedback or requested changes from maintainers.

**I am fine with requiring rods instead (should be either 2 or 4) and a few more bottles about.** Anything past that I think we should rather be considering avenues to make supplying the fob a more streamlined/enjoyable process and or reducing the quantity/likelihood of bone breaks. For example a working chute between req and the hangar and maybe a gel_refiller also in req. **I am also fine with different kinds of bone break surgeries using different amounts of bone gel like Beagle was contemplating internally. @BeagleGaming1 address what you like from above and I think everything is good to go.**

**Comment by Cooliecan** [comment]
Could you perhaps consider changing bone gel so treating hands/feet uses 2%, arms and legs 3% and chest and head 5%? It's too taxing having everything as 5% and it makes sense to use less bone gel for smaller areas.

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: Mapping good with new changes

**Review by Nanu308 — APPROVED** [review]
get code approval and mergeable

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by nauticall — APPROVED** [review]
sproites

**Pull request merged**
